Lib/logging/__init__.py

```diff
@@ -883,25 +875,20 @@ def _removeHandlerRef(wr):
     # set to None. It can also be called from another thread. So we need to
     # pre-emptively grab the necessary globals and check if they're None,
     # to prevent race conditions and failures during interpreter shutdown.
-    acquire, release, handlers = _acquireLock, _releaseLock, _handlerList
-    if acquire and release and handlers:
-        acquire()
-        try:
-            handlers.remove(wr)
-        except ValueError:
-            pass
-        finally:
-            release()
+    handlers = _handlerList
```

Comment:
Right, Python shutdown sets all module global variables to None. Please copy _lock as lock (local variable).

Reply:
Updated to assign `_lock` to local variable `lock`.